### PR TITLE
chore: [CDS-69660]: allow custom notes for pipeline execution and aborted pipelines

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

Changelist:
 - User can add notes from the execution page header (`View Note` / `Add a New Note`)
 - From the deployments/execution history page: New note can be added from the Kebab menu `Add Execution Notes`
 - If notes are present - the note icon will have a `View Note` option in the popover to open the view/edit dialog
 - Notes can also be edited/added from the `Abort Pipeline` confirmation dialog
 - Aborted status has a popover to indicate abortedBy information

TODO:
- Comment icon visibility will be based on API response boolean ( support yet to be added )
- Execution page header `View Note/Add a note` text will be based on API response boolean ( support yet to be added )

https://github.com/harness/harness-core-ui/assets/100121280/ba8d37a1-dfb8-45e9-a986-29d1c30d160c

image
image
image
image
image
image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vikrant-harness — DISMISSED** [review]

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 1]
```diff
@@ -1923,3 +1924,9 @@ downloadLogs:
   downloadRequestFailed: 'Download failed with: {{ errorMsg }}'
 infrastructurePropagatedFrom: 'Infrastructure Propagated From:'
 viewInCompiledMode: 'View in Compiled Mode'
+viewNote: View Note
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:1927]: nit: these all can be clubbed into one common parent string constant.

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+describe('useNotesModal test', () => {
+  const modal = findDialogContainer()
+  test('render dialog with NotesModalForm - assert update and discard and close modal', async () => {
+    const { container, getByText, getByPlaceholderText, findByText } = render(
+      <TestWrapper>
+        <WrapperComponent />
+      </TestWrapper>
+    )
+
+    const openModal = container.querySelector('.openModal')
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx:74]: nit: We can give labels to buttons and find buttons using that.

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/NotesModalForm.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            resourceScope: {
+              accountIdentifier: accountId,
+              orgIdentifier,
+              projectIdentifier
+            },
+            resource: {
+              resourceType: ResourceType.PIPELINE,
+              resourceIdentifier: pipelineExecutionSummary?.pipelineIdentifier
+            },
+            permission: PermissionIdentifier.EXECUTE_PIPELINE
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/NotesModalForm.tsx:110]: I think permission here should change based on where you are. In pipeline studio, you might want to check edit permission whereas in execution view, you should be checking execute permission. No ?

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    await act(async () => {
+      fireEvent.click(openModal!)
+    })
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx:77]: remove act wrap

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    <>
+      <button className="openModal" onClick={onBtnClick} />
+      <button className="updateNotesModal" onClick={onUpdate} />
+      <div className="notesContent">{notes}</div>
+    </>
+  )
+}
+
+describe('useNotesModal test', () => {
+  const modal = findDialogContainer()
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx:66]: find the modal inside tests

**Review by rameshx — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx (hunk 2 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/__tests__/useNotesModal.test.tsx:75]: This class was added in the mockRenderer for this hook only to test

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/NotesModalForm.tsx (hunk 3 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/NotesModalForm.tsx:129]: Notes currently are only present and linked with executions, so added RBAC for execution permissions after discussing with PM.

**Review by vikrant-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**